### DISCUSSION TITLE
Add download-url-suffix to generate_appcast

### DISCUSSION THREAD

**Discussion by brzzdev** [sparkle-project/Sparkle#2637] [Q&A] [answered]
## Summary

For debug builds of our app, they're behind our internal GitLab instance and require a `private_key`. The `feedParameters` of `SPUUpdaterDelegate` works wonderfully for grabbing the XML itself, but then the DMG fails to download.

Right now, I have to edit the XML after creation to append `?private_key=exampleKey` to the URL. Having a built-in option to match the `download-url-prefix` would be amazing.

## Version

All

**Comment by zorgiepoo** [comment]
Custom HTTP headers are passed to the appcast & download. I think auth tokens and such normally use headers, not feed parameters. Adding parameters on the URL for this seems odd.

**Reply by brzzdev** [reply, marked as answer]
Yeah, turns out I was using `feedParameters` when I should have been using `willDownloadUpdate item with request`. I didn't realise the latter existed until your answer pointed me in the right direction.

**Reply by zorgiepoo** [reply]
I actually forgot about that delegate method, that would work. Though I think having private key in the feed parameters is still odd.
